Port documentation from Wiki: ShortText Field #332 (#340)
SenseNet/sensenet.github.io: _docs/shorttext-field.md

title:  "ShortText Field"
source_url: 'https://github.com/SenseNet/sensenet/blob/master/docs/shorttext-field.md'
category: Concepts
version: v7.0.0
tags: [fields, shorttext]
description: ShortText Field is used for storing short strings. Maximum length of the stored data is 450 characters (this value however can be limited to a lower value in the Field Setting.
---

# ShortText Field
 
## Overview

> ShortText [Field](http://wiki.sensenet.com/Field) is used for storing short strings. Maximum length of the stored data is 450 characters (this value however can be limited to a lower value in the Field Setting of the Field). If you want to store text strings longer than this consider using the LongText Field.

Reviewer: ```suggestion
ShortText [Field](/field) is used for storing short strings. Maximum length of the stored data is 450 characters (this value however can be limited to a lower value in the Field Setting of the Field). If you want to store text strings longer than this consider using the [LongText Field](/longtext-field).
```

## Details

The following apply to the behavior of the Field:

> - **Regular expression**: data input can be validated against a regular expression defined in the [Field Settings](http://wiki.sensenet.com/Field_Setting).

Reviewer: ```suggestion
- **Regular expression**: data input can be validated against a regular expression defined in the [Field Settings](/field-setting).
```

## Field handler
- handler: SenseNet.ContentRepository.Fields.ShortTextField
- short name: **ShortText**

Usage in CTD:
```xml
   <Field name="ShortDesc" type="ShortText">
   ...
   </Field>
```

## Supported Field Controls

> - [ShortText Field Control](http://wiki.sensenet.com/ShortText_Field_Control): a simple textbox control where user can edit value (if non-readonly).

Reviewer: ```suggestion
- [ShortText Field Control](/shorttext-fieldcontrol): a simple textbox control where user can edit value (if non-readonly).
```

## Configuration

> The following properties can be set in the Field's [Field Setting](http://wiki.sensenet.com/Field_Setting) configuration:

Reviewer: ```suggestion
The following properties can be set in the Field's [Field Setting](/field-setting) configuration:
```

      </Configuration>
</Field>
```
The above example configures the ShortText Field so that:

- default value is set to hello world
- at most 15 characters are allowed to be inserted
- at least 3 characters are expected to be inserted
- Field value is editable (not read-only)
- Field can contain only letters and numbers
## Related links
- [Field](http://wiki.sensenet.com/Field)
- [ShortText Field Control](http://wiki.sensenet.com/ShortText_Field_Control)

> ## References

Reviewer: please remove this part
